src/widgets/QDisassemblyView.cpp

```diff
@@ -1285,38 +1286,85 @@ void QDisassemblyView::paintEvent(QPaintEvent *) {
 			return a.distance < b.distance;
 		});
 
-		int width_upward_outviewport = 0;
-		int width_downward_outviewport = 0;
+		// find suitable arrow horizontal width
+		for (int i = 0; i < jump_arrow_vec.size(); i++) {
 
-		for (const auto& jump_arrow : jump_arrow_vec) {
+			JumpArrow& jump_arrow = jump_arrow_vec[i];
+			bool is_dst_upward = jump_arrow.target < instructions_[jump_arrow.src_line].rva();
 
-			// get current process state
-			State state;
-			IProcess* process = edb::v1::debugger_core->process();
-			process->current_thread()->get_state(&state);
+			int size_block = font_width_ * 2;
 
-			bool is_dst_upward = jump_arrow.target < instructions_[jump_arrow.src_line].rva();
-			int arrow_horizontal_length;
-			
-			if (jump_arrow.dst_in_viewport) {
+			for (int j = size_block; j < l1; j += size_block) {
 
-				arrow_horizontal_length = jump_arrow.distance * (font_width_ / 3);
+				bool is_width_good = true;
 
-			} else if (is_dst_upward) {
+				// check if width clash with previous jump arrow
+				for (int k = 0; k < i; k++) {
 
-				arrow_horizontal_length = width_upward_outviewport + font_width_ * 2;
-				width_upward_outviewport = arrow_horizontal_length;  // update new width
+					JumpArrow& jump_arrow_prev = jump_arrow_vec[k];
+					bool is_dst_upward_prev = jump_arrow_prev.target < instructions_[jump_arrow_prev.src_line].rva();
 
-			} else {
+					int jump1_min = std::min(
+						jump_arrow.src_line, 
+						jump_arrow.dst_in_viewport ? jump_arrow.dst_line : (is_dst_upward ? 0 : viewport()->height())
+					);
+
+					int jump1_max = std::max(
+						jump_arrow.src_line, 
+						jump_arrow.dst_in_viewport ? jump_arrow.dst_line : (is_dst_upward ? 0 : viewport()->height())
+					);
+
+					int jump2_min = std::min(
+						jump_arrow_prev.src_line, 
+						jump_arrow_prev.dst_in_viewport ? jump_arrow_prev.dst_line : (is_dst_upward_prev ? 0 : viewport()->height())
+					);
+					
+					int jump2_max = std::max(
+						jump_arrow_prev.src_line,
+						jump_arrow_prev.dst_in_viewport ? jump_arrow_prev.dst_line : (is_dst_upward_prev ? 0 : viewport()->height())
+					);
+
+					bool cond1 = jump2_max >= jump1_max && jump2_min >= jump1_max;
+					bool cond2 = jump2_min <= jump1_min && jump2_max <= jump1_min;
+					bool is_jump_overlap = !(cond1 || cond2);	
+
+					if (is_jump_overlap && 
+						j == jump_arrow_prev.horizontal_width) {
```

Comment:
Maybe avoid splitting this line? It seems short enough when merged (especially compared to the ternary conditionals above).

Reply:
Fixed.